docs/conceptual/More-about-how-ROCm-uses-PCIe-Atomics.rst

```diff
@@ -1,156 +1,63 @@
-signaled to the requester by setting the Completion Status field in the completion descriptor, they are
-set to to Completer Abort (CA) or Unsupported Request (UR).
+Working with PCIe 3.0 in ROCm
+-------------------------------
+Starting with PCIe 3.0, atomic operations can be requested, routed through, and completed by PCIe components. Routing and completion do not require software support. Component support for each can be identified by the Device Capabilities 2 (DevCap2) register. Upstream
+bridges need to have atomic operations routing enabled. If not enabled, the atomic operations will fail even if the 
+PCIe endpoint and PCIe I/O devices can perform atomic operations. 
 
-To understand more about how PCIe atomic operations work, see
-`PCIe atomics <https://pcisig.com/specifications/pciexpress/specifications/ECN_Atomic_Ops_080417.pdf>`_
+If your system uses PCIe Express Switch to connect and enable communication between multiple PCIe components, the switch must also support atomic operations routing.
 
-`Linux Kernel Patch to pci_enable_atomic_request <https://patchwork.kernel.org/project/linux-pci/patch/1443110390-4080-1-git-send-email-jay@jcornwall.me/>`_
+To enable atomic operations routing between multiple root ports, each root port must support atomic operation routing. This capability can be identified from the atomic operations routing supported bit in the DevCap2 register. If the bit has value of 1, routing is supported. Atomic operation requests are permitted only if a component's ``DEVCTL2.ATOMICOP_REQUESTER_ENABLE``
+field is set. These requests can only be serviced if the upstream components also support atomic operation
+completion, or can route it to a component that supports it, or both.  
 
-There are also a number of papers which talk about these new capabilities:
+ROCm uses the PCIe-ID-based ordering technology for peer-to-peer (P2P) data transmission. PCIe-ID-based ordering technology is used when the GPU initiates multiple write operations to different memory locations. For example. when there are two write operations which need to be run in this order::
 
-  * `Atomic Read Modify Write Primitives by Intel <https://www.intel.es/content/dam/doc/white-paper/atomic-read-modify-write-primitives-i-o-devices-paper.pdf>`_
-  * `PCI express 3 Accelerator White paper by Intel <https://www.intel.sg/content/dam/doc/white-paper/pci-express3-accelerator-white-paper.pdf>`_
-  * `PCIe Generation 4 Base Specification includes atomic operations <https://astralvx.com/storage/2020/11/PCI_Express_Base_4.0_Rev0.3_February19-2014.pdf>`_
-  * `Xilinx PCIe Ultrascale White paper <https://docs.xilinx.com/v/u/8OZSA2V1b1LLU2rRCDVGQw>`_
+1. Write to another GPU memory.
+2. Write to system memory to indicate the transfer is complete.
 
-Other I/O devices with PCIe atomics support:
+The two write operations are routed to different locations. However, use of the ordering technology ensures that the order of the operation is maintained. 
```

Comment:
This is not always true because our implementation is buggy and broken. There are HW blocks in our GPU that break PCIe ordering rules. It may be better to leave this out unless you want to go into excruciating detail.